server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -93,48 +101,32 @@ public void start(Settings settings, TransportService transportService, ClusterS
         }
 
         final IncrementalClusterStateWriter incrementalClusterStateWriter
-            = new IncrementalClusterStateWriter(settings, clusterService.getClusterSettings(), metaStateService,
+            = new IncrementalClusterStateWriter(settings, clusterSettings, metaStateService,
                 manifestClusterStateTuple.v1(),
-                prepareInitialClusterState(transportService, clusterService, manifestClusterStateTuple.v2()),
+                prepareInitialClusterState(transportService, clusterSettings, manifestClusterStateTuple.v2()),
                 transportService.getThreadPool()::relativeTimeInMillis);
         if (DiscoveryNode.isMasterNode(settings) == false) {
```

Comment:
Suggest inverting this `if` -- it fixes the line length issue here and also improves harmony. Also my comment should have read `master-eligible nodes` not `master-ineligible nodes`.

```suggestion
        if (DiscoveryNode.isMasterNode(settings)) {
            // Master-eligible nodes must persist the cluster state when accepting it because they must reload the (complete, fresh)
            // last-accepted cluster state when restarted.
            persistedState.set(new MasterEligibleNodePersistedState(incrementalClusterStateWriter));
        } else if (DiscoveryNode.isDataNode(settings)) {
            // Master-ineligible data nodes blah blah...
            persistedState.set(new DataOnlyNodePersistedState(settings, incrementalClusterStateWriter, transportService.getThreadPool()));
        } else {
            // Non-master non-data nodes do not need to persist the cluster state as they do not use a persistent store
            persistedState.set(new InMemoryPersistedState(manifestClusterStateTuple.v1().getCurrentTerm(), manifestClusterStateTuple.v2()));
        }
```